File: skill_sdk/tests/fixtures/java-project/src/main/java/com/fixture/shop/service/OrderService.java
package com.fixture.shop.service;

import java.util.List;

import com.fixture.shop.cart.Cart;
import com.fixture.shop.discount.DiscountStrategy;
import com.fixture.shop.discount.DiscountUtils;
import com.fixture.shop.models.Order;
import com.fixture.shop.models.OrderStatus;
import com.fixture.shop.payment.PaymentGateway;
import com.fixture.shop.repository.Repository;

/**
 * Facade over the checkout flow — top-level entry point for incomingCalls.
 */
public class OrderService {

    private final Repository repo;
    private final PaymentGateway gateway;
    private final List<DiscountStrategy> strategies;

    public OrderService(Repository repo, PaymentGateway gateway, List<DiscountStrategy> strategies) {
        this.repo = repo;
        this.gateway = gateway;
        this.strategies = strategies;
    }

    /**
     * Full order flow — outgoingCalls target.
     * Calls: cart.toOrder, DiscountUtils.applyBestDiscount, cart.checkout.
     */
    public Order placeOrder(Cart cart, String orderId) {
        if (!strategies.isEmpty()) {
            Order order = cart.toOrder(orderId, null);
            DiscountUtils.applyBestDiscount(order, strategies);
            return cart.checkout(orderId, repo, gateway, strategies.get(0));
        }
        return cart.checkout(orderId, repo, gateway, null);
    }

    /**
     * Cancel an order and update its status.
     * Calls: repo.getOrder, repo.updateOrderStatus.
     */
    public void cancelOrder(String orderId) {
        Order order = repo.getOrder(orderId);
        if (order == null) {
            return;
        }
        order.setStatus(OrderStatus.CANCELLED);
        repo.updateOrderStatus(orderId, OrderStatus.CANCELLED);
    }
}

File: skill_sdk/tests/fixtures/ts-project/src/shop/service/OrderService.ts
import { Cart } from '../cart/Cart';
import { DiscountStrategy } from '../discount/DiscountStrategy';
import { DiscountUtils } from '../discount/DiscountUtils';
import { Order } from '../models/Order';
import { OrderStatus } from '../models/OrderStatus';
import { PaymentGateway } from '../payment/PaymentGateway';
import { Repository } from '../repository/Repository';

/**
 * Facade over the checkout flow — top-level entry point for incomingCalls.
 */
export class OrderService {
  constructor(
    private readonly repo: Repository,
    private readonly gateway: PaymentGateway,
    private readonly strategies: DiscountStrategy[]
  ) {}

  /**
   * Full order flow — outgoingCalls target.
   * Calls: cart.toOrder, DiscountUtils.applyBestDiscount, cart.checkout.
   */
  placeOrder(cart: Cart, orderId: string): Order {
    if (this.strategies.length > 0) {
      const order = cart.toOrder(orderId, null);
      DiscountUtils.applyBestDiscount(order, this.strategies);
      return cart.checkout(orderId, this.repo, this.gateway, this.strategies[0]);
    }
    return cart.checkout(orderId, this.repo, this.gateway, null);
  }

  /**
   * Cancel an order and update its status.
   * Calls: repo.getOrder, repo.updateOrderStatus.
   */
  cancelOrder(orderId: string): void {
    const order = this.repo.getOrder(orderId);
    if (!order) {
      return;
    }
    order.setStatus(OrderStatus.CANCELLED);
    this.repo.updateOrderStatus(orderId, OrderStatus.CANCELLED);
  }
}
